pkg/macaroon/serialize.go

```diff
@@ -0,0 +1,205 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package macaroon
+
+import (
+	"encoding/binary"
+	"errors"
+)
+
+type fieldType int
```

Comment:
Do we expect negative numbers?
If not, I personally use `uint`.

I personally always prefer to specify the size rather than defaulting to some that change depending of the underlying architecture, but that's another story.

Reply:
Below, I've seen that used `-1` as an invalid value, so `uint` cannot be if that's not changed.
I would still consider to use `uint` and use `0` as an invalid value.

However, I haven't read the Macaroons spec, so I don't know if it must be in this way.